ui/app/hooks/useTransactionDisplayData.js

```diff
@@ -0,0 +1,169 @@
+import { useSelector } from 'react-redux'
+import { getKnownMethodData } from '../selectors/selectors'
+import { getTransactionActionKey, getStatusKey } from '../helpers/utils/transactions.util'
+import { camelCaseToCapitalize } from '../helpers/utils/common.util'
+import { useI18nContext } from './useI18nContext'
+import { PRIMARY, SECONDARY } from '../helpers/constants/common'
+import { getTokenToAddress } from '../helpers/utils/token-util'
+import { useUserPreferencedCurrency } from './useUserPreferencedCurrency'
+import { formatDateWithYearContext, shortenAddress } from '../helpers/utils/util'
+import {
+  APPROVED_STATUS,
+  SUBMITTED_STATUS,
+  CONTRACT_INTERACTION_KEY,
+  DEPLOY_CONTRACT_ACTION_KEY,
+  INCOMING_TRANSACTION_KEY,
+  TOKEN_METHOD_TRANSFER,
+  TOKEN_METHOD_TRANSFER_FROM,
+  SEND_ETHER_ACTION_KEY,
+  TRANSACTION_CATEGORY_APPROVAL,
+  TRANSACTION_CATEGORY_INTERACTION,
+  TRANSACTION_CATEGORY_RECEIVE,
+  TRANSACTION_CATEGORY_SEND,
+  TRANSACTION_CATEGORY_SIGNATURE_REQUEST,
+  PENDING_STATUS,
+  TOKEN_METHOD_APPROVE,
+} from '../helpers/constants/transactions'
+import { useCurrencyDisplay } from './useCurrencyDisplay'
+import { useTokenDisplayValue } from './useTokenDisplayValue'
+import { useTokenData } from './useTokenData'
+import { tokenSelector } from '../selectors'
+
+// This is duplicated from transactions selectors, additionally
+// from the history view we will already know the pending status
+// of a transaction, but in order to make this hook less coupled
+// to where it is used we should determine the pending status here
+const pendingStatusMap = {
+  [APPROVED_STATUS]: true,
+  [SUBMITTED_STATUS]: true,
+}
+
+const tokenCategoryMap = {
+  [TOKEN_METHOD_APPROVE]: true,
```

Comment:
Yes, it is used to show the spend limit which it has done historically as well. The original submission prior to changes from feedback worked to display the spend limit, but once I added the explicit bypassing of token related variables e2e tests started failing. What had happened was exactly what you were hoping to prevent or surface : the token related variables were being used in places I hadn’t expected and when I became more explicit about passing them in the approve workflow broke. So this was added to be able to correctly use the token data/displayValue for approval

Reply:
Ah I see. I was assuming that we _didn't_ want to show the spend limit in the transaction list, as the new design kinda implies only the ETH cost is shown? But it is ambiguous, and this is how it works already, so this seems fine for now. We can ask for clarity on this later. I think you asked a similar question in a Figma comment.